linkerd/app/outbound/src/sidecar.rs

```diff
@@ -376,6 +384,12 @@ impl svc::Param<watch::Receiver<tls::Routes>> for TlsSidecar {
     }
 }
 
+impl svc::Param<AllowHostnameLabels> for TlsSidecar {
```

Comment:
In my mind sticking that to the target allows for easier way to eventually rely on discovery information for configuring that options. Correct me if I am wrong here.